### DISCUSSION TITLE
HTTP DELETE with BODY

### DISCUSSION THREAD

**Discussion by magnayn** [micronaut-projects/micronaut-core#11045] [General] [answered]
The keycloak REST API, sadly, uses "DELETE with BODY". There's no point in pointing at specs that say you shouldn't do this - people have and that's just a fact of life.

I cannot find a way of working around the fact that micronaut-http does not like that, and fails to include the body I provide. I think this is down fundamentally to HttpRequestFactory.builder, which is just statically invoked from AbstractJdkHttpClient.

(I'm using an @Client).

How can I work around this?

**Comment by yawkat** [comment, marked as answer]
Sounds like a bug in micronaut-http-client-jdk. I created #11047 to track
